modules/maps.js

```diff
@@ -233,29 +213,34 @@ class MappingProfile {
             } else if (game.upgrades.Dominance.done) {
                 ratio /= 4;
             }
-            // Stop increasing map level once we get to the right ratio
-            if (this.z <= 40 && ratio > 1.5) break;
-            if (ratio > 1.2) break;
+            // Stop increasing map level once HD ratio is too large
+            if ((this.z <= 40 && ratio > 1.5) || ratio > 1.2) {
+                break;
+            }
         }
         // Keep increasing map level while we can overkill
         const extraMapLevelsAvailable = this.hze >= 209;
-        if (extraMapLevelsAvailable && this.optimalLevel === this.baseLevel) {
-            const oneShotPower = maxOneShotPower();
-            while (oneShotZone(this.optimalLevel + 1, "map", "S") === oneShotPower) {
-                this.optimalLevel++;
-            }
-        }
-        if (game.global.challengeActive !== "Coordinate" && !mutations.Magma.active()) {
-            // Prefer "Oneshot level" + 1, except on magma or in Coordinated challenge
-            this.optimalLevel++;
-        }
         if (!extraMapLevelsAvailable) {
             // can't increase map levels yet
             this.optimalLevel = Math.min(this.optimalLevel, this.z);
+        } else {
+            if (extraMapLevelsAvailable && this.optimalLevel >= this.baseLevel) {
+                const maxOneShotCells = maxOneShotPower();
+                while (oneShotZone(this.optimalLevel + 1, "map", "S") >= maxOneShotCells) {
+                    this.optimalLevel++;
+                }
+            }
+            if (game.global.challengeActive !== "Coordinate" && !mutations.Magma.active()) {
+                // Prefer "Oneshot level" + 1, except on magma or in Coordinated challenge
+                this.optimalLevel++;
+            }
         }
-        if (this.haveMapReducer && (this.optimalLevel === this.z) && (this.minLevel <= this.baseLevel)) {
-            // if min level allows, and we have Map Reducer, lower the map level
-            this.optimalLevel = this.baseLevel;
+        // if we have Map Reducer and there's nothing useful in the optimal map level, consider lowering it
+        if (this.haveMapReducer
+            && (this.optimalLevel === this.z)
+            && (this.minLevel <= this.baseLevel)
+            && !mapLevelHasPrestiges(this.optimalLevel)) {
+                this.optimalLevel = this.baseLevel;
         }
```

Comment:
Could you please change this if style?
I'd prefer if you used some variables to keep the if condition clause in the same line, or something like that =X

Reply:
Done